core/Pos/Arbitrary/Crypto.hs

```diff
@@ -62,61 +49,33 @@ instance Arbitrary SignTag where
 -- Arbitrary signing keys
 ----------------------------------------------------------------------------
 
--- If you want an arbitrary keypair, just generate a secret key with
--- 'arbitrary' and then use 'Pos.Crypto.toPublic' to get the corresponding
--- public key.
-
-keys :: [(PublicKey, SecretKey)]
-keys = unsafeMakePool "[generating keys for tests...]" 50 keyGen
-{-# NOINLINE keys #-}
-
 instance Arbitrary PublicKey where
-    arbitrary = fst <$> elements keys
+    arbitrary = fst . deterministicKeyGen <$> arbitrarySizedS 32
```

Comment:
The reason for using Secure Random™ was that it seemed better to test *exactly* the functions that are used for generating keys (e.g. what if `keyGen` puts additional constraints on the seed and just doing `arbitrarySizedS 32` isn't enough?). I still think that it's better, but having deterministic tests is better still.